app/src/main/java/com/github/kulebin/myfoursquareapp/http/HttpClient.java

```diff
@@ -17,18 +17,27 @@
     private static final int READ_TIMEOUT = 5000;
     private static final int CONNECTION_TIMEOUT = 8000;
 
+    private IInterceptor mInterceptor;
+
     @Override
     public void doRequest(final HttpRequest pHttpRequest, final IOnResult pIOnResult) {
-        execute(pHttpRequest.getUrl(),
-                pHttpRequest.getRequestType(),
-                pHttpRequest.getHeaders(),
-                pHttpRequest.getBody(),
+        final HttpRequest httpRequest = modifyRequest(pHttpRequest);
+        execute(httpRequest.getUrl(),
+                httpRequest.getRequestType(),
+                httpRequest.getHeaders(),
+                httpRequest.getBody(),
                 pIOnResult);
     }
 
     @Override
     public void doRequest(final String pUrl, final IOnResult pIOnResult) {
-        execute(pUrl, HttpRequestType.GET, null, null, pIOnResult);
+        final String url = modifyRequest(pUrl);
```

Comment:
move modifyRequest logic to execute, let execute method accept HttpRequest as argument. 
It will help you not to forget about modifications when you add any other doRequest methods